idea/idea-completion/src/org/jetbrains/kotlin/idea/completion/handlers/handlerUtils.kt

```diff
@@ -144,8 +146,14 @@ fun createKeywordConstructLookupElement(
                     val beforeCaret = tailBeforeCaret.indentLinesAfterFirst(newIndent)
                     val afterCaret = tailAfterCaret.indentLinesAfterFirst(newIndent)
 
-                    insertionContext.document.insertString(offset, beforeCaret + afterCaret)
-                    insertionContext.editor.moveCaret(offset + beforeCaret.length)
+                    val element = insertionContext.file.findElementAt(offset)?.getNextSiblingIgnoringWhitespaceAndComments(true)
+
+                    if (element != null && beforeCaret.trimStart().startsWith(insertionContext.document.getText(TextRange.from(element.startOffset, 1)))) {
```

Comment:
The reason I used `startsWith()` is that some keywords insert more than a single character, e.g. `init` where `beforeCaret.trimStart()` will be equal to "{\n".

See https://github.com/JetBrains/kotlin/blob/b5db50f42996c09de12a1bddb98adb960a32bc01/idea/idea-completion/src/org/jetbrains/kotlin/idea/completion/KeywordCompletion.kt#L72-L72